src/functions-and-arrays.js

```diff
@@ -1,19 +1,116 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 >= num2) {
+    return num1;
+  } else if (num1 <= num2) {
+    return num2;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(arr) {
+  if (arr.length === 0) {
+    return null
+  } else if (arr.length === 1) {
+    return arr[0];
+  } else {
+    let longestWord = '';
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i].length > longestWord.length) {
+        longestWord = arr[i]
+      }
+    }
+    return longestWord;
+  }
+}
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(arr) {
+  if (arr.length === 0) {
+    return 0;
+  } else {
+    let sum = 0;
+    for (let i = 0; i < arr.length; i++) {
+      sum += arr[i];
+    }
+    return sum;
+  }
+}
+
+function sum(arr) {
+  if (arr.length === 0) {
+    return 0;
+  } else {
+    let sum = 0;
+    for (let i = 0; i < arr.length; i++) {
+      if (typeof arr[i] === 'number' || typeof arr[i] === 'boolean') {
+        sum += arr[i];
+      } else if (typeof arr[i] === 'string') {
+        sum += arr[i].length;
+      } else {
+        throw new Error("Unsupported data type sir or ma'am");
+      }
+    }
+    return sum;
+  }
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+  if (arr.length === 0) {
+    return null;
+  } else {
+    let sum = 0;
+    for (let i = 0; i < arr.length; i++) {
+      sum += arr[i];
+    }
+    let average = sum / arr.length;
+    return average;
+  }
+}
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
+function averageWordLength(arr) {
+  if (arr.length === 0) {
+    return null;
+  } else {
+    let sum = 0;
+    for (let i = 0; i < arr.length; i++) {
+      sum += arr[i].length
+    }
+    let average = sum / arr.length;
```

Comment:
Will the variable "average" be reassigned. If not, how about declaring a const instead of a let?